File: sample-rest-api/src/main/java/nl/haploid/octowight/sample/controller/AbstractController.java
package nl.haploid.octowight.sample.controller;

import nl.haploid.octowight.sample.data.ResourceNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;

public class AbstractController {

	@ExceptionHandler(ResourceNotFoundException.class)
	@ResponseStatus(HttpStatus.NOT_FOUND)
	public String handleResourceNotFoundException() {
		return "Resource not found";
	}
}

File: sample-rest-api/src/main/scala/nl/haploid/octowight/sample/controller/AbstractController.scala
package nl.haploid.octowight.sample.controller

import nl.haploid.octowight.sample.data.ResourceNotFoundException
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.{ExceptionHandler, ResponseStatus}

class AbstractController {

  @ExceptionHandler(Array(classOf[ResourceNotFoundException]))
  @ResponseStatus(HttpStatus.NOT_FOUND)
  def handleResourceNotFoundException = "Resource not found"
}
